libs/MobileSync/src/com/salesforce/androidsdk/mobilesync/app/MobileSyncUpgradeManager.java

```diff
@@ -81,4 +95,12 @@ protected synchronized void upgradeSObject() {
     public String getInstalledSobjectVersion() {
         return getInstalledVersion(MOBILE_SYNC_KEY);
     }
+
+    /*
+     * TODO: Remove this in Mobile SDK 9.0.
+     */
+    private void upgradeTo8Dot2() {
+        LayoutSyncManager.getInstance().getSmartStore().dropAllSoups();
```

Comment:
Why remove all the soups? Should you simply delete the soups that layout sync sank into?
Alternatively, could you write the query to work with the old keys and new keys for layouts?

Reply:
Ah yes, you're right. Copy-paste error from the tests ;-) I will tweak this to drop just the `layout` soup.

Reply:
Done [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2045/commits/cf1f6224e34d2e313726b94e1452f9b07dd0cb9f).